icons/src/fullcolor/render-bitmaps-verbose.py

```diff
@@ -238,60 +221,278 @@ def characters(self, chars):
 
     # If invocation does not include a file name, process all SVGs in listed sources
     else:
-        print("Rendering from path \"" + SRC + "\"")
-        if not os.path.exists(OUTPUT_DIRECTORY):
-            os.mkdir(OUTPUT_DIRECTORY)
+        print("Rendering from path \"" + category_to_render + "\"")
+        if not os.path.exists(args.output_path):
+            os.mkdir(args.output_path)
 
-        for file in os.listdir(SRC):
+        # this is the loop for the files in the given directory
+        for file in os.listdir(category_to_render):
             if file[-4:] == '.svg':
-                file = os.path.join(SRC, file)
-                handler = ContentHandler(file)
+                file = os.path.join(category_to_render, file)
+                #
+                # ContentHandler() implements an interface for xml.sax.parse()
+                #
+                handler = ContentHandler(file, args)
+                #
+                # xml.sax.parse() uses the various ContentHandler() methods behind the scenes
+                #
                 xml.sax.parse(open(file), handler)
         
         return True
 
-    
 
-parser = argparse.ArgumentParser(description='Render icons from SVG to PNG')
+def check_dependencies(args):
```

Comment:
using https://docs.python.org/3.9/library/shutil.html#shutil.which this really simplifies in something like

```python
for dependency in ["inkscape", "optipng"]:
    if not shutil.which(dependency):
        print(f"Please install {dependency}")
```

which in my system results in
`Please install optipng`